unlock-js/src/__tests__/deploy.test.js

```diff
@@ -114,29 +40,46 @@ describe('contract deployer', () => {
     status: '0x1',
   }
   beforeEach(() => {
-    accountsAndYield(unlockAccountsOnNode)
-    ethGasPriceAndYield()
-    sendTransactionAndYield(unlockAccountsOnNode[0], transaction.hash)
-    ethGetTransactionReceiptAndYield(transaction.hash, transactionReceipt)
-    ethGetCodeAndYield('0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', 'latest')
-    ethGasPriceAndYield()
-    sendTransactionAndYield(
-      unlockAccountsOnNode[0],
-      transaction.hash,
-      '0xc4d66de8000000000000000000000000aaadeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // data (no idea what this is, but there it is)
-      '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // contract address
-      '0xf4240' // gas
+    nock.accountsAndYield(unlockAccountsOnNode)
+    nock.ethGasPriceAndYield(gasPrice)
+    // contract deploy call
+    nock.ethSendTransactionAndYield(
+      {
+        from: unlockAccountsOnNode[0],
+        data: Unlock.bytecode,
+        gas: '0x' + GAS_AMOUNTS.deployContract.toString(16),
+      },
+      gasPrice,
+      transaction.hash
+    )
+    nock.ethGetTransactionReceipt(transaction.hash, transactionReceipt)
+    // get the contract bytecode
```

Comment:
One test we need to do is see how the web3Service behaves when this yields `0x` (it should throw) This is a key test because if the contract is not deployed, nothing will succeed after that so we would want to know about failures.

Reply:
👍added new test